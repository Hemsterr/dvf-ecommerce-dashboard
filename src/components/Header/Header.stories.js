// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { BrowserRouter as Router } from 'react-router-dom'

// Components
import Header from './index'

storiesOf('Header', module).add('Default', () => (
  <Router>
    <Header primary />
  </Router>
))
