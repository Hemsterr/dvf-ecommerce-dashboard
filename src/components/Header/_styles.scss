.main-header {
  background-color: $shuttle-gray;
  padding: pxToRem(22px);
  display: flex;
  justify-content: space-between;
  align-items: center;

  & img {
    width: pxToRem(216px);
  }

  &__link {
    @include text-size(pxToRem(12px), pxToRem(3.6px), pxToRem(16px));
    color: $alabaster;
    text-transform: uppercase;
    font-family: $font-base-bold;
  }
}

@media only screen and (max-width: $screen-sm) {
  .main-header {
    justify-content: center;

    &__link {
      display: none;
    }
  }
}

@media only screen and (max-width: $screen-xs) {
  .main-header {
    padding: pxToRem(12px);
    position: fixed;
    width: 100%;
    z-index: 100;

    & img {
      width: pxToRem(108px);
    }
  }
}
