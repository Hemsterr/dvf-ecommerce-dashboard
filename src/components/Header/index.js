// @flow
import React from 'react'
import { Link } from 'react-router-dom'

// Themes
import images from '../../themes/images'

// Constants
import ROUTES from '../../constants/routes'

type Props = {
  primary: Boolean,
}

const Header = ({ primary }: Props) => (
  <header className="main-header">
    <h1>
      <Link to={ROUTES.FITTING_OPTIONS}>
        <img src={images.hemster} alt="Hemster" />
      </Link>
    </h1>
    {primary && (
      <Link to={ROUTES.FITTING_OPTIONS} className="main-header__link">
        back to fitting options
      </Link>
    )}
  </header>
)

Header.defaultProps = {
  primary: false,
}

export default Header
