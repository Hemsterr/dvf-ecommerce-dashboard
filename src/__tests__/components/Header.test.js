// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'

// Component
import Header from '../../components/Header'

const mockProps = {
  primary: true,
}

const wrapper = (props = {}) => {
  const component = shallow(<Header {...props} />)

  return component
}

describe('Header component', () => {
  const wrapperWithoutProps = wrapper()

  // Snapshot
  test('should render correctly Header component default props', () => {
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
    expect(Header.defaultProps.primary).toBeFalsy()
  })

  // Testing props
  test('should be rendered default props', () => {
    expect(
      wrapperWithoutProps
        .find('img')
        .first()
        .props().src
    ).toBeTruthy()
    expect(wrapperWithoutProps.find('.main-header__link').length).toBe(0)
  })

  test('should be rendered with props', () => {
    const { container } = render(
      <Router>
        <Header {...mockProps} />
      </Router>
    )
    expect(container.querySelectorAll('.main-header__link').length).toEqual(1)
  })
})
