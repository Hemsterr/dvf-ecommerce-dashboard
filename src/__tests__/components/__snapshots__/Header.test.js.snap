// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly Header component default props 1`] = `
<header
  className="main-header"
>
  <h1>
    <Link
      to="/fitting-options"
    >
      <img
        alt="Hemster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </Link>
  </h1>
</header>
`;
